tests/pairedomics/test_antismash_downloader.py

```diff
@@ -0,0 +1,245 @@
+
+    assert (archive1.exists() and archive2.exists())
+    assert (archive1.is_file() and archive2.is_file())
+    assert (extracted_folder1.exists() and extracted_folder2.exists())
+    assert all(
+        Path(extracted_folder1, extracted_file).is_file()
+        for extracted_file in extracted_files1)
+    assert all(
+        Path(extracted_folder2, extracted_file).is_file()
+        for extracted_file in extracted_files2)
+    assert genome_status_file.is_file()
+    assert len(genome_status) == 2
+
+
+def test_missing_id(tmp_path):
+    download_root = tmp_path / "download"
+    extract_root = tmp_path / "extracted"
+    genome_status_file = Path(download_root, "genome_status.csv")
+    genome_records = [{
+        "genome_ID": {
+            "genome_type": "genome",
+            "JGI_Genome_ID": "",
+            "RefSeq_accession": ""
+        },
+        "genome_label": "Salinispora arenicola CNX508"
+    }, {
+        "genome_ID": {
+            "genome_type": "genome",
+            "JGI_Genome_ID": "2515154177",
+            "RefSeq_accession": "GCF_000514515.1"
+        },
+        "genome_label": "Salinispora pacifica CNT029"
+    }]
+
+    podp_download_and_extract_antismash_data(genome_records, download_root,
+                                             extract_root)
+
+    archive1 = download_root / "GCF_000514875.1.zip"
+    extracted_folder1 = extract_root / "antismash" / "GCF_000514875.1"
+    archive2 = download_root / "GCF_000514515.1.zip"
+    extracted_folder2 = extract_root / "antismash" / "GCF_000514515.1"
+    genome_status = _get_genome_status_log(genome_status_file)
+
+    assert (not archive1.exists() and archive2.exists())
+    assert (not archive1.is_file() and archive2.is_file())
+    assert (not extracted_folder1.exists() and extracted_folder2.exists())
+    assert genome_status_file.is_file()
+    assert len(genome_status) == 1
+
+
+def test_caching(tmp_path):
+    download_root = tmp_path / "download"
+    extract_root = tmp_path / "extracted"
+    genome_status_file = Path(download_root, "genome_status.csv")
+    genome_records = [{
+        "genome_ID": {
+            "genome_type": "genome",
+            "JGI_Genome_ID": "2515154188",
+            "RefSeq_accession": "GCF_000514875.1"
+        },
+        "genome_label": "Salinispora arenicola CNX508"
+    }]
+
+    podp_download_and_extract_antismash_data(genome_records, download_root,
+                                             extract_root)
+    genome_status_old = _get_genome_status_log(genome_status_file)
+    genome_obj = genome_status_old["GCF_000514875.1"]
+    assert Path(genome_obj.filename).exists()
+    assert genome_obj.attempted
+    podp_download_and_extract_antismash_data(genome_records, download_root,
+                                             extract_root)
+    genome_status_new = _get_genome_status_log(genome_status_file)
+    assert len(genome_status_old) == len(genome_status_new)
```

Comment:
These tests are not serving the purpose `test_caching`. No matter the function use the caching or not, the old and new genome status is always same for the same input. 
I did not find a way to test the caching for the current code. If you really want to test that, it's better to insert an logging when it's using cache and then test the logging.

Reply:
@hechth

Reply:
Great!